rstudio/rstudio
# allow offline RPC execution when running notebook chunks

### Intent

Addresses https://github.com/rstudio/rstudio-pro/issues/5607.

### Approach

Allow for the offline RPC thread to execute RPCs while notebook chunks are executing.

### Automated Tests

N/A

### QA Notes

Test via notes in https://github.com/rstudio/rstudio-pro/issues/5607. You can also test with this sample document:

`````

```{r setup, include=FALSE}
library(nycflights13)
library(DBI)
conn <- dbConnect(RSQLite::SQLite())
dbWriteTable(conn, "flights", flights)
```


```{sql connection=conn}
SELECT *, *, *, *, *, *, *, *,
       *, *, *, *, *, *, *, *,
       *, *, *, *, *, *, *, *
       FROM flights
```

```{r}
dbDisconnect(conn)
```
`````

Try executing the SQL chunk, and then editing and saving a file while the query runs. You should still be able to successfully do so.

### Documentation

N/A

### Checklist
